bsip-0056.md

```diff
@@ -0,0 +1,314 @@
+- when a proposal is done for a managed vesting balance, and `delay_seconds > 0` is true, the funds are not immediately 
+transferred to the owner, but will be automatically when `delay_seconds` seconds have passed
+ - The specified `amount` can be greater than the current possible maximum of the vesting balance. In such a case, the maximum is withdrawn instead of the specified `amount`
+- The `amount` can be set to `0`, which will cancel any pending withdrawal (triggered through a former withdrawal operation and `delay_seconds > 0`)
+
+## Deposit Operation
+
+Users of the new vesting balance policy will need to be able to also deposit funds into a vesting balance. 
+To allow for this a new deposit operation is implemented
+
+    struct vesting_balance_deposit_operation : public base_operation
+    {
+       asset                       fee;
+       account_id_type             fee_paying_account;
+       vesting_balance_id_type     vesting_balance;
+       asset                       deposit_amount;
+
+       extension< ext > extensions;
+    };
+
+Anyone can deposit into any vesting balance, as long as it is the same asset.
+
+ - the `fee_paying_account` must hold `deposit_amount`, and this will be deducted from the corresponding balance
+ - the `owner` of the `vesting_balance` will receive `deposit_amount`
+
+## Annulment of vesting balances
+
+This new type of vesting balance must be able to know when it was `last used` (created, deposited, withdrawn, settled). This 
+timestamp is updated every time the vesting balance is used. A new committee parameter is introduced: `cleanup_unused_vesting_balances_after_seconds` (seconds). If the vesting balance was unused for longer than `cleanup_unused_vesting_balances`, it is annuled during maintenance interval. This applies only to managed vesting balances.
+
+Additional checks during maintenance:
+ - Query all unused managed vesting balances
+   - If it has a balance greater than 0, liquidate by forcing a withdrawal (just as if the user would manually initiate a withdrawal), no fee payment necessary. This action may or may not update the `last used` time (developers choice to what is feasible). Deletion of this object will not happen within this maintenance interval
+   - If the balance is 0, delete the vesting balance object, no return of any fees
+
+## Honoring White and Blacklisting
+
+Managed vesting balances must account for white- and blacklisted assets and accounts, and behave in that manner as if the asset would be held directly by the account.
+
+Vesting Balances can only be created or charged (deposit) if
+- the owner and the fee paying account could also directly hold the contained asset (black- and whitelist check)
+- the owner of the vesting balance has not blacklisted (or whitelisted) the fee paying account and/or the manager
+
+Vesting Balances can only be settled if 
+- the owner and the fee paying account also directly hold the contained asset (black- and whitelist check)
```

Comment:
The new owner (after the settlement) or the old owner (before the settlement)?  Both?

Reply:
In a settlement both accounts would have their vesting balances settled, so for both the checks must hold.